src/main/java/io/vertx/openapi/contract/impl/MediaTypeImpl.java

```diff
@@ -30,7 +30,23 @@ public MediaTypeImpl(String identifier, JsonObject mediaTypeModel) {
     this.mediaTypeModel = mediaTypeModel;
     JsonObject schemaJson = mediaTypeModel.getJsonObject(KEY_SCHEMA);
     if (schemaJson == null || schemaJson.isEmpty()) {
-      throw createUnsupportedFeature("Media Type without a schema");
+      // Inject default value if schema is left out
+      // by using shortcut "application/octet-stream: {}"
+      // See https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0
+      if (identifier.equalsIgnoreCase(MediaType.APPLICATION_OCTET_STREAM)) {
+        // In OpenAPI v3.0, describing file uploads is signalled with a type:
+        // string and the format set to byte, binary, or base64.
+        // In OpenAPI v3.1, JSON Schema helps make this far more clear with
+        // its contentEncoding and contentMediaType keywords,
+        // which are designed for exactly this sort of use.
+        // See https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-8.6
+        schemaJson = new JsonObject()
+          .put("type", "string")
+          .put("format", "binary")
+          .put("contentMediaType", "application/octet-stream");
```

Comment:
This should become a constant maybe in `MediaType`. I would also remove `contentMediaType` as in the [example](https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0) only format and type is mentioned. This is then also compatible with 3.0

Reply:
I'll replace it as a constant in `MediaType`.

But stripping the `contentMediaType` isn't a good idea, because it replaces `format` in OpenAPI 3.1. I've intentionally placed both keys in that JSON object to be compatible with OpenAPI 3.x.

That way it's future proof, because later adaptions of this library for OpenAPI 3.1 or later might break.

Am I overthinking that part?

Reply:
Mh maybe you are right, but I think we can simply ignore `contentMediaType `, because the MediaType is already specified with `if (identifier.equalsIgnoreCase(MediaType.APPLICATION_OCTET_STREAM)) {`

> JSON Schema also offers a contentMediaType keyword. However, when the media type is already specified by the Media Type Object's key, or by the contentType field of an [Encoding Object](https://swagger.io/specification/#encoding-object), the contentMediaType keyword SHALL be ignored if present.

Would you agree?

Reply:
> This should become a constant maybe in `MediaType`. I would also remove `contentMediaType` as in the [example](https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0) only format and type is mentioned. This is then also compatible with 3.0

If someone uses the alias `application/octet-stream: {}`, it's safe to assume that the contract file is of version 3.1, because that wasn't allowed before in 3.0 and lower. Therefore I'm removing `format` from the `schemaJson`.
The final schema object that gets injected into the media type object would look like the following:
```
new JsonObject()
          .put("type", "string")
          .put("contentMediaType", "application/octet-stream");
```

> Mh maybe you are right, but I think we can simply ignore contentMediaType , because the MediaType is already specified with `if (identifier.equalsIgnoreCase(MediaType.APPLICATION_OCTET_STREAM)) {`

Yeah, that makes sense to ignore it here.

Summary: For contract validation we place a default schema object like seen above, if someone uses the alias definition introduced in OpenAPI 3.1. For request validation we simplify the validation if the media type object ID is `application/octet-stream`.

Sounds good?

Reply:
What's the benefit of having `contentMediaType` in the default schema object? I still don't understand it.

Why can't we simply use

```
new JsonObject()
          .put("type", "string")
          .put("format", "binary")
```

?

Reply:
**The default schema object must only be applied if someone has an `openapi-contract.yml` file with OpenAPI 3.1. Before that, no shortcuts like `{}` could be used in the schema file.**

In OpenAPI 3.0 you must write:
```
# an arbitrary binary file:
content:
    application/octet-stream:
        schema:
            type: string
            format: binary
```
In OpenAPI 3.1 you (can) now write:
```
# a PNG image as a binary file:
content:
    image/png: {}
# an arbitrary binary file:
content:
    application/octet-stream: {}
```
And that gets expanded to:
```
# a PNG image as a binary file:
content:
    image/png:
        schema:
            type: string
            contentMediaType: image/png
            contentEncoding: base64
# an arbitrary binary file:
content:
    application/octet-stream:
        schema:
            type: string
            contentMediaType: application/octet-stream
```
It really is confusing! As of my understanding, in OpenAPI 3.1, `format` has been replaced by `contentMediaType` when dealing with `Schema Object`s in relation to binary file uploads. See https://spec.openapis.org/oas/latest.html#considerations-for-file-uploads

The task of this library must be the expansion before the validation of the contract. Right now I'm only providing that expansion for `application/octet-stream`, because everything else isn't part of this PR.

We could still use your approach of the old way of declaring it, but maybe that will create confusion in later stages of development of this library? Supporting multiple versions of OpenAPI is challenging.

Reply:
Later in the code, if necessary, you can distinguish between 3.0 and 3.1 only be checking the keys of the schema object. If `format` is missing but `contentMediaType` is present, it must be 3.1. Otherwise it will be 3.0 schema. This might be come in handy...

Reply:
Sorry I don't get the specs .... maybe we don't need a schema object at all. It seems like it is intentionally left out when using `{}` in the contract.

Then it is treated like
> Content transferred in binary (octet-stream)

OK, so leave out the schema object? What do you think?